.github/workflows/ci.yml

```diff
@@ -0,0 +1,42 @@
+name: CI
+on: [push, pull_request]
+env:
+  CI: true
+  SCALA_VERSION: 2.13.1
+jobs:
+  validate:
+    name: Scala ${{ matrix.scala }}, Java ${{ matrix.java }}
+    runs-on: ${{matrix.systems}} 
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [adopt@1.8, adopt@1.11, adopt@1.14,graalvm@]
+        scala: [2.12.11, 2.13.1]
+        #TODO: windows-latest likely requires another [job](https://github.com/olafurpg/setup-scala)
+        systems: [ubuntu-latest, macos-latest, windows-latest]
+    env:
+      SCALA_VERSION: ${{ matrix.scala }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: olafurpg/setup-scala@v7
+        with:
+          java-version: ${{ matrix.java }}
+      # TODO: should we remove?
```

Comment:
I think maybe this one should be caching by a hash of the sbt files, like the next one.  From the [Github guide](https://github.com/actions/cache/blob/master/examples.md#scala---sbt):

```yaml
- name: Cache SBT ivy cache
  uses: actions/cache@v1
  with:
    path: ~/.ivy2/cache
    key: ${{ runner.os }}-sbt-ivy-cache-${{ hashFiles('**/build.sbt') }}
- name: Cache SBT
  uses: actions/cache@v1
  with:
    path: ~/.sbt
    key: ${{ runner.os }}-sbt-${{ hashFiles('**/build.sbt') }}
```

Just replace the ivy references with coursier.

Reply:
I'd be better off to read but there's plenty of stuff to do. Thanks!